sysbuild: Add SB_OVERLAY_CONFIG

57300 opened the pull request:
This fixes a minor issue in sysbuild, where `-DOVERLAY_CONFIG=...` would be applied not only to the main application, but also sysbuild itself. This was incorrect, because sysbuild imports a different Kconfig tree than normal Zephyr builds, so the same Kconfig fragment file would not necessarily be valid for both.

This adds a new variable SB_OVERLAY_CONFIG to resolve this ambiguity. It functions along the lines of SB_CONF_FILE, with both being sysbuild-specific versions of existing variables.

To ensure that OVERLAY_CONFIG is still passed on to the main application verbatim, its value is now loaded in `configuration_files.cmake`, rather than `kconfig.cmake`. This is because the former file is not imported by sysbuild, and it is where the related variables, such as CONF_FILE and DTC_OVERLAY_FILE, are loaded as well.

nordicjm reviewed (commented):
Not sure this is an issue, overlay files are not useful for sysbuild, but configuration files could be used to configure sysbuild itself for a particular configuration. Will await @tejlmand's input on this

tejlmand commented on share/sysbuild/cmake/modules/sysbuild_kconfig.cmake:
```diff
@@ -48,6 +48,7 @@ file(TOUCH ${CMAKE_CURRENT_BINARY_DIR}/empty.conf)
 set(APPLICATION_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR})
 set(AUTOCONF_H             ${CMAKE_CURRENT_BINARY_DIR}/autoconf.h)
 set(CONF_FILE              ${SB_CONF_FILE})
+set(OVERLAY_CONFIG         "${CMAKE_CURRENT_BINARY_DIR}/empty.conf")
```
we should probably support a `SB_OVERLAY_CONFIG` for sysbuild.

```suggestion
set(OVERLAY_CONFIG         "${SB_OVERLAY_CONFIG}")
```

Note, quoting is used to ensure that `OVERLAY_CONFIG` as minimum becomes an empty string, cause otherwise the local scope `OVERLAY_CONFIG` would become undefined and thus the cache variable will be used instead (which we don't want in this case).

57300 replied:
Sounds good. I wasn't sure if `SB_OVERLAY_CONFIG` would be needed, but I can add it here.

tejlmand commented on cmake/modules/configuration_files.cmake:
```diff
@@ -11,6 +11,8 @@
 #
 # - CONF_FILE:              List of Kconfig fragments
+# - OVERLAY_CONFIG:         List of additional Kconfig fragments
+#                           (to be mixed in with CONF_FILE)
```
I think there is a general understanding of how `OVERLAY_CONFIG` works.
Let's just remove this part.
```suggestion
```

tejlmand reviewed (changes requested):
Some minor comments.

> This fixes a minor issue in sysbuild, where `-DOVERLAY_CONFIG=...` would
not be applied to the main application, but rather sysbuild itself. 

I don't believe this is accurate. `OVERLAY_CONFIG` would be applied to the main application, but also applied to sysbuild (which this commit is fixing).

So please adjust to:
```
This fixes a minor issue in sysbuild, where `-DOVERLAY_CONFIG=...` would
be applied to both sysbuild and the main application. 
```

57300 committed:
sysbuild: Add SB_OVERLAY_CONFIG

This fixes a minor issue in sysbuild, where `-DOVERLAY_CONFIG=...` would
be applied not only to the main application, but also sysbuild itself.
This was incorrect, because sysbuild imports a different Kconfig tree
than normal Zephyr builds, so the same Kconfig fragment file would not
necessarily be valid for both.

This adds a new variable SB_OVERLAY_CONFIG to resolve this ambiguity.
It functions along the lines of SB_CONF_FILE, with both being sysbuild-
specific versions of existing variables.

To ensure that OVERLAY_CONFIG is still passed on to the main application
verbatim, its value is now loaded in `configuration_files.cmake`, rather
than `kconfig.cmake`. This is because the former file is not imported by
sysbuild, and it is where the related variables, such as CONF_FILE and
DTC_OVERLAY_FILE, are loaded as well.

Signed-off-by: Grzegorz Swiderski <grzegorz.swiderski@nordicsemi.no>

57300 commented:
> I don't believe this is accurate. `OVERLAY_CONFIG` would be applied to the main application, but also applied to sysbuild (which this commit is fixing).

True, thanks for catching this. I rewrote the commit message to explain the issue better; please check if it's correct.

tejlmand reviewed (approved):
lgtm, thanks for fixing this.
